--- a/pam.c
+++ b/pam.c
@@ -280,11 +280,7 @@
 			errx(1, "Authentication required");
 
 		/* doas style prompt for pam */
-		char host[HOST_NAME_MAX + 1];
-		if (gethostname(host, sizeof(host)))
-			snprintf(host, sizeof(host), "?");
-		snprintf(doas_prompt, sizeof(doas_prompt),
-		    "\rdoas (%.32s@%.32s) password: ", myname, host);
+		snprintf(doas_prompt, sizeof(doas_prompt), "\rPassword: ");
 
 		/* authenticate */
 		ret = pam_authenticate(pamh, 0);
--- a/shadow.c
+++ b/shadow.c
@@ -74,11 +74,7 @@
 		errx(1, "Authentication failed");
 	}
 
-	char host[HOST_NAME_MAX + 1];
-	if (gethostname(host, sizeof(host)))
-		snprintf(host, sizeof(host), "?");
-	snprintf(cbuf, sizeof(cbuf),
-			"\rdoas (%.32s@%.32s) password: ", myname, host);
+	snprintf(cbuf, sizeof(cbuf), "\rPassword: ");
 	challenge = cbuf;
 
 	response = readpassphrase(challenge, rbuf, sizeof(rbuf), RPP_REQUIRE_TTY);
